### ISSUE TITLE
RVX Manager aborts when patching YTM v6.11.52

### ISSUE THREAD

**Issue by KobeW50** [inotia00/ReVanced_Extended#1191]
### Type

Error while patching

### Tools used

ReVanced Manager

### Application

YT Music v6.11.52

With RVX Manager 1.4.0

### Bug description

Manager aborts with exit code 1 when patching ytm 6.11.52 with the default patches.

Tested with 6.10.51 and 6.08.51 and those worked.

### Steps to reproduce

Patch 6.11.52 with the default patches and see it abort.

### Relevant log output

```shell
Initializing installer
Creating working directory
Copying original apk
Unpacking input apk
Reading dex files
Decoding AndroidManifest.xml only, because resources are not needed
Merging integrations
Deleting existing resource cache directory
Decoding resources
Executing patches
Applied Amoled
Applied Background play
Applied Bitrate default value
Applied Certificate spoof
Applied Custom branding icon Revancify blue
Applied Custom package name
Applied Disable auto captions
Applied Enable black navigation bar
Applied Enable color match player
Applied Enable compact dialog
Applied Enable custom filter
Applied Enable dismiss queue
Applied Enable force minimized player
Applied Enable force shuffle
Applied Enable landscape mode
Applied Enable minimized playback
Applied Enable new layout
Applied Enable old style miniplayer
Applied Enable opus codec
Applied Enable sleep timer
Applied Enable zen mode
Applied Exclusive audio playback
Applied Hide button shelf
Applied Hide carousel shelf
Applied Hide cast button
Applied Hide category bar
Applied Hide get premium
Applied Hide music ads
Applied Hide navigation label
Applied Hide new playlist button
Applied Hide playlist card
Applied Hide taste builder
Applied Hide upgrade button
Applied Microg support
Applied Optimize resource
Applied Remember video quality
Applied Settings
Applied Share button hook
Applied Spoof app version
Applied Translations
Repacking patched apk
Compiling resources
An error occurred! Aborted
Error:
brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [/data/app/~~j2SW0npZ2vePnBexQrZpwA==/app.rvx.manager.flutter--tTsTsUDcnykX-JXuK8EpQ==/lib/arm64/libaapt2.so, link, -o, /data/user/0/app.rvx.manager.flutter/cache/patcher/tmp-PZORRG/cache/aapt_temp_file, --min-sdk-version, 21, --target-sdk-version, 33, --version-code, 61152240, --version-name, 6.11.52, --no-auto-version, --no-version-vectors, --no-version-transitions, --no-resource-deduping, --allow-reserved-package-id, -e, /data/user/0/app.rvx.manager.flutter/cache/APKTOOL756504361920709998.tmp, -0, arsc, -I, /data/user/0/app.rvx.manager.flutter/cache/patcher/tmp-PZORRG/cache/1.apk, --manifest, /data/user/0/app.rvx.manager.flutter/cache/patcher/tmp-PZORRG/cache/AndroidManifest.xml, /data/user/0/app.rvx.manager.flutter/cache/patcher/tmp-PZORRG/cache/build/resources.zip]
	at brut.androlib.res.AndrolibResources.aapt2Package(AndrolibResources.java:484)
	at brut.androlib.res.AndrolibResources.aaptPackage(AndrolibResources.java:619)
	at app.revanced.patcher.Patcher.save(Patcher.kt:128)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda-20(MainActivity.kt:217)
	at app.revanced.manager.flutter.MainActivity.$r8$lambda$Rd-IpIH_jmxD15EWmBEFM-jtGsk(Unknown Source:0)
	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda13.run(Unknown Source:24)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: brut.common.BrutException: could not exec (exit code = 1): [/data/app/~~j2SW0npZ2vePnBexQrZpwA==/app.rvx.manager.flutter--tTsTsUDcnykX-JXuK8EpQ==/lib/arm64/libaapt2.so, link, -o, /data/user/0/app.rvx.manager.flutter/cache/patcher/tmp-PZORRG/cache/aapt_temp_file, --min-sdk-version, 21, --target-sdk-version, 33, --version-code, 61152240, --version-name, 6.11.52, --no-auto-version, --no-version-vectors, --no-version-transitions, --no-resource-deduping, --allow-reserved-package-id, -e, /data/user/0/app.rvx.manager.flutter/cache/APKTOOL756504361920709998.tmp, -0, arsc, -I, /data/user/0/app.rvx.manager.flutter/cache/patcher/tmp-PZORRG/cache/1.apk, --manifest, /data/user/0/app.rvx.manager.flutter/cache/patcher/tmp-PZORRG/cache/AndroidManifest.xml, /data/user/0/app.rvx.manager.flutter/cache/patcher/tmp-PZORRG/cache/build/resources.zip]
	at brut.util.OS.exec(OS.java:99)
	at brut.androlib.res.AndrolibResources.aapt2Package(AndrolibResources.java:480)
	... 6 more
```

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.
- [X] I have written the title and contents in English.
- [X] This issue does not reproduce on unpatched YouTube or YT Music.

**Comment by KobeW50** [comment]
An error also occurs in Revancify (for this YTM version)

```
INFO: Reading dex files 
INFO: Decoding AndroidManifest.xml only, because resources are not needed 
INFO: Skipping custom-branding-music-name: Excluded by default 
INFO: Skipping custom-branding-icon-mmt: Excluded by default 
INFO: Skipping custom-branding-icon-revancify-red: Excluded by default 
INFO: Skipping enable-debug-logging: Excluded by default 
INFO: Merging inotia00-integrations-v0.112.3.apk 
INFO: Deleting existing resource cache directory 
INFO: Decoding resources 
INFO: Executing patches 
INFO: Amoled succeeded 
INFO: Background play succeeded 
INFO: Bitrate default value succeeded 
INFO: Certificate spoof succeeded 
INFO: Custom branding icon Revancify blue succeeded 
INFO: Custom package name succeeded 
INFO: Disable auto captions succeeded 
INFO: Enable black navigation bar succeeded 
INFO: Enable color match player succeeded 
INFO: Enable compact dialog succeeded 
INFO: Enable custom filter succeeded 
INFO: Enable dismiss queue succeeded 
INFO: Enable force minimized player succeeded 
INFO: Enable force shuffle succeeded 
INFO: Enable landscape mode succeeded 
INFO: Enable minimized playback succeeded 
INFO: Enable new layout succeeded 
INFO: Enable old style miniplayer succeeded 
INFO: Enable opus codec succeeded 
INFO: Enable sleep timer succeeded 
INFO: Enable zen mode succeeded 
INFO: Exclusive audio playback succeeded 
INFO: Hide button shelf succeeded 
INFO: Hide carousel shelf succeeded 
INFO: Hide cast button succeeded 
INFO: Hide category bar succeeded 
INFO: Hide get premium succeeded 
INFO: Hide music ads succeeded 
INFO: Hide navigation label succeeded 
INFO: Hide new playlist button succeeded 
INFO: Hide playlist card succeeded 
INFO: Hide taste builder succeeded 
INFO: Hide upgrade button succeeded 
INFO: Microg support succeeded 
INFO: Optimize resource succeeded 
INFO: Remember video quality succeeded 
INFO: Settings succeeded 
INFO: Share button hook succeeded 
INFO: Spoof app version succeeded 
INFO: Translations succeeded 
INFO: Compiling resources 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6262: error: resource android:color/bright_foreground_dark_disabled is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6265: error: resource android:color/btn_default_material_dark is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6266: error: resource android:color/btn_leanback_color is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6267: error: resource android:color/bright_foreground_dark_inverse is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6268: error: resource android:color/btn_default_material_light is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6269: error: resource android:color/btn_leanback_focused is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6270: error: resource android:color/background_floating_device_default_light is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6271: error: resource android:color/background_device_default_light is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6272: error: resource android:color/car_action1 is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6273: error: resource android:color/car_action1_dark is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6274: error: resource android:color/background_holo_light is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6275: error: resource android:color/background_floating_material_light is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6276: error: resource android:color/car_blue_100 is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6277: error: resource android:color/car_blue_200 is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6278: error: resource android:color/bright_foreground_disabled_holo_light is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6279: error: resource android:color/accent_device_default_700 is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6280: error: resource android:color/btn_colored_borderless_text_material is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6281: error: resource android:color/bright_foreground_dark is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6282: error: resource android:color/background_material_dark is private. 
WARNING: /data/data/com.termux/files/home/revancify-data/revanced-cache/res/values/styles.xml:6283: error: resource android:color/car_blue_50 is private. 
brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [./aapt2, link, -o, /data/data/com.termux/files/home/revancify-data/revanced-cache/aapt_temp_file, --min-sdk-version, 21, --target-sdk-version, 33, --version-code, 61152240, --version-name, 6.11.52, --no-auto-version, --no-version-vectors, --no-version-transitions, --no-resource-deduping, --allow-reserved-package-id, -e, /data/data/com.termux/files/usr/tmp/APKTOOL6736845185901419365.tmp, -0, arsc, -I, revanced-cache/1.apk, --manifest, /data/data/com.termux/files/home/revancify-data/revanced-cache/AndroidManifest.xml, /data/data/com.termux/files/home/revancify-data/revanced-cache/build/resources.zip]
	at brut.androlib.res.AndrolibResources.aapt2Package(AndrolibResources.java:484)
	at brut.androlib.res.AndrolibResources.aaptPackage(AndrolibResources.java:619)
	at app.revanced.patcher.Patcher.save(Patcher.kt:128)
	at app.revanced.cli.patcher.Patcher.start$revanced_cli(Patcher.kt:21)
	at app.revanced.cli.command.MainCommand.run(MainCommand.kt:173)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at app.revanced.cli.main.MainKt.main(Main.kt:7)
Caused by: brut.common.BrutException: could not exec (exit code = 1): [./aapt2, link, -o, /data/data/com.termux/files/home/revancify-data/revanced-cache/aapt_temp_file, --min-sdk-version, 21, --target-sdk-version, 33, --version-code, 61152240, --version-name, 6.11.52, --no-auto-version, --no-version-vectors, --no-version-transitions, --no-resource-deduping, --allow-reserved-package-id, -e, /data/data/com.termux/files/usr/tmp/APKTOOL6736845185901419365.tmp, -0, arsc, -I, revanced-cache/1.apk, --manifest, /data/data/com.termux/files/home/revancify-data/revanced-cache/AndroidManifest.xml, /data/data/com.termux/files/home/revancify-data/revanced-cache/build/resources.zip]
	at brut.util.OS.exec(OS.java:99)
	at brut.androlib.res.AndrolibResources.aapt2Package(AndrolibResources.java:480)
	... 13 more



Variant: nonRoot
Arch: arm64-v8a
App: YouTube-Music v6.11.52
CLI: inotia00-cli-v2.22.1.jar
Patches: inotia00-patches-v2.184.3.jar
Integrations: inotia00-integrations-v0.112.3.apk
Patches argument: -i amoled
-i background-play
-i bitrate-default-value
-i certificate-spoof
-i custom-branding-icon-revancify-blue
-i custom-package-name
-i disable-auto-captions
-i enable-black-navigation-bar
-i enable-color-match-player
-i enable-compact-dialog
-i enable-custom-filter
-i enable-dismiss-queue
-i enable-force-minimized-player
-i enable-force-shuffle
-i enable-landscape-mode
-i enable-minimized-playback
-i enable-new-layout
-i enable-old-style-miniplayer
-i enable-opus-codec
-i enable-sleep-timer
-i enable-zen-mode
-i exclusive-audio-playback
-i hide-button-shelf
-i hide-carousel-shelf
-i hide-cast-button
-i hide-category-bar
-i hide-get-premium
-i hide-music-ads
-i hide-navigation-label
-i hide-new-playlist-button
-i hide-playlist-card
-i hide-taste-builder
-i hide-upgrade-button
-i microg-support
-i optimize-resource
-i remember-video-quality
-i settings
-i share-button-hook
-i spoof-app-version
-i translations
```

**Comment by inotia00** [comment]
Google added a new M3 color of Android 14+ in YouTube v18.27.34+ and YT Music v6.11.52+

Currently, the ApkTool used in revanced-patcher does not have information about the M3 color of Android 14, so it cannot be decoded. This causes errors

Bumping ApkTool is the only solution, but ApkTool is speculated that it will take a considerable time to reflect it because it contains a huge change between last year and today

**Comment by KobeW50** [comment]
> Bumping ApkTool is the only solution, but ApkTool is speculated that it will take a considerable time to reflect it because it contains a huge change between last year and today

Damn.

I reported this issue to the ReVanced team, and oSumAtrix said I should open the issue in the patcher repo. Are you planning on opening an issue there as well? If yes, you should do it instead of me bec you know 100x more than me.

**Comment by KobeW50** [comment]
Also, if it's gonna take a while for the solution, will you change the suggested version of YTM?

**Comment by inotia00** [comment]
Local repository (my PC) is already preparing to support it

Perhaps next week will provide support for YouTube v18.27.34 and YT Music v6.11.52

**Comment by Adity1243** [comment]
> Google added a new M3 color of Android 14+ in YouTube v18.27.34+ and YT Music v6.11.52+
> 
> Currently, the ApkTool used in revanced-patcher does not have information about the M3 color of Android 14, so it cannot be decoded. This causes errors
> 
> Bumping ApkTool is the only solution, but ApkTool is speculated that it will take a considerable time to reflect it because it contains a huge change between last year and today

Does that M3 colour works on android 12 devices or it will only work on android 14+ devices ?

**Comment by inotia00** [comment]
m3 color is related to Android 14+, but in order to compile YouTube, androlib must be able to decode M3 color

**Comment by inotia00** [comment]
fixed in revanced-patches-v2.186.1
